lib/cucumber/runtime.rb

```diff
@@ -67,6 +67,7 @@ def run!
       receiver = Test::Runner.new(@configuration.event_bus)
       compile features, receiver, filters
       @configuration.notify :test_run_finished
+      fire_after_all_hook
```

Comment:
This should be moved up one line, so that the `test_run_finished` event means "all userland code that will be executed has been executed".